### PR TITLE
Adding --json support for status command

### PR DESCRIPTION
**Pull request by slater0013**
New Edit from #532
Implements #531

### CODE REVIEW

File: src/renault_api/cli/renault_vehicle.py [hunk 1]
```diff
@@ -153,13 +154,17 @@ async def display_status(
...
-    await update_res_state(vehicle, status_table)
-    await update_hvac_status(vehicle, status_table)
+    await update_battery_status(vehicle, status_table, ctx_data))
+    await update_charge_mode(vehicle, status_table, ctx_data))
+    await update_cockpit(vehicle, status_table, ctx_data))
+    await update_location(vehicle, status_table, ctx_data))
+    await update_lock_status(vehicle, status_table, ctx_data))
+    await update_res_state(vehicle, status_table, ctx_data))
+    await update_hvac_status(vehicle, status_table, ctx_data))
+
```
Comment (epenet) [inline src/renault_api/cli/renault_vehicle.py:164]: Remove the space here
```suggestion

```

File: src/renault_api/cli/renault_vehicle.py [hunk 2]
```diff
@@ -153,13 +154,17 @@ async def display_status(
...
-    await update_lock_status(vehicle, status_table)
-    await update_res_state(vehicle, status_table)
-    await update_hvac_status(vehicle, status_table)
+    await update_battery_status(vehicle, status_table, ctx_data))
+    await update_charge_mode(vehicle, status_table, ctx_data))
+    await update_cockpit(vehicle, status_table, ctx_data))
+    await update_location(vehicle, status_table, ctx_data))
+    await update_lock_status(vehicle, status_table, ctx_data))
+    await update_res_state(vehicle, status_table, ctx_data))
+    await update_hvac_status(vehicle, status_table, ctx_data))
```
Comment (epenet) [inline src/renault_api/cli/renault_vehicle.py:163]: Remove trailing brackets here
```suggestion
    await update_battery_status(vehicle, status_table, ctx_data)
    await update_charge_mode(vehicle, status_table, ctx_data)
    await update_cockpit(vehicle, status_table, ctx_data)
    await update_location(vehicle, status_table, ctx_data)
    await update_lock_status(vehicle, status_table, ctx_data)
    await update_res_state(vehicle, status_table, ctx_data)
    await update_hvac_status(vehicle, status_table, ctx_data)
```

Comment (slater0013) [reply src/renault_api/cli/renault_vehicle.py:163]: copy/paste fail

File: src/renault_api/cli/renault_vehicle.py [hunk 3]
```diff
@@ -5,6 +5,7 @@
 from typing import Optional
 from typing import Tuple
 
+import json
```
Comment (epenet) [inline src/renault_api/cli/renault_vehicle.py:8]: Please move this to the top of the file, having it here breaks CI.

**Comment by epenet** [comment]
Great - now it's just the test that needs adding

**Comment by epenet** [comment]
I think you will need to duplicate this test:
Take a look at this: https://github.com/hacf-fr/renault-api/blob/13c784b6af09331368341c93888f1eb32c46cb19/tests/cli/test_vehicle.py#L255-L257

**Comment by slater0013** [comment]
Something like def test_vehicle_status_json_no_prompt( ?

**Comment by epenet** [comment]
> Something like def test_vehicle_status_json_no_prompt( ?

Yes, or simply `test_vehicle_status_json`

**Comment by slater0013** [comment]
Ok I hope the json EXPECTED_STATUS is well formated...

File: tests/cli/test_vehicle.py [hunk 4]
```diff
@@ -129,6 +129,13 @@
         "HVAC status        on\n"
         "-----------------  -------------------------\n"
     ),
+    "zoe_40.1_json.json": (
+        "{\"battery-status\": {\"timestamp\": \"2020-11-17T09:06:48+01:00\", \"batteryLevel\": 50, \"batteryAutonomy\": 128, \"batteryCapacity\": 0, \"batteryAvailableEnergy\": 0, \"plugStatus\": 0, \"chargingStatus\": -1.0}, "
+        "\"charge-mode\": {\"chargeMode\": \"always\"}, "
+        "\"cockpit\": {\"totalMileage\": 49114.27 }, "
+        "\"res-state\": {\"details\": \"Stopped, ready for RES\", \"code\": \"10\"}, "
+        "\"hvac-status\": {\"externalTemperature\": 8.0, \"hvacStatus\": \"off\"}}"
+    ),
```
Comment (epenet) [inline tests/cli/test_vehicle.py:138]: Very close... one extra space and one carriage return missing at the end.
I've also split it to keep under about 88 chars long.
```suggestion
    "zoe_40.1_json.json": (
        '{"battery-status": {"timestamp": "2020-11-17T09:06:48+01:00", '
        '"batteryLevel": 50, "batteryAutonomy": 128, "batteryCapacity": 0, '
        '"batteryAvailableEnergy": 0, "plugStatus": 0, "chargingStatus": -1.0}, '
        '"charge-mode": {"chargeMode": "always"}, '
        '"cockpit": {"totalMileage": 49114.27}, '
        '"res-state": {"details": "Stopped, ready for RES", "code": "10"}, '
        '"hvac-status": {"externalTemperature": 8.0, "hvacStatus": "off"}}\n'
    ),
```

File: tests/cli/test_vehicle.py [hunk 5]
```diff
@@ -273,6 +282,27 @@ def test_vehicle_status_no_prompt(
 
     assert EXPECTED_STATUS["zoe_40.1.json"] == result.output
 
+def test_vehicle_status_json(
```
Comment (epenet) [inline tests/cli/test_vehicle.py:285]: Missing carriage return here.
```suggestion

def test_vehicle_status_json(
```

File: src/renault_api/cli/renault_vehicle.py [hunk 6]
```diff
@@ -296,6 +314,9 @@ async def update_lock_status(
         click.echo(f"lock status: {exc.error_details}", err=True)
         return
 
+    if ctx_data["json"]:
```
Comment (epenet) [inline src/renault_api/cli/renault_vehicle.py:317]:
```suggestion
    if ctx_data["json"]: # pragma: no cover
```

File: src/renault_api/cli/renault_vehicle.py [hunk 7]
```diff
@@ -272,6 +287,9 @@ async def update_location(
         click.echo(f"location: {exc.error_details}", err=True)
         return
 
+    if ctx_data["json"]:
```
Comment (epenet) [inline src/renault_api/cli/renault_vehicle.py:290]:
```suggestion
    if ctx_data["json"]: # pragma: no cover
```

**Review by epenet — COMMENTED** [review]
Coverage

**Comment by epenet** [comment]
Thanks @slater0013 👍

**Comment by epenet** [comment]
To make it easier to contribute some more in the future, I suggest that you install `VSCode` and enable `formatting on save` with the `black` plugin.
This will avoid all the spacing issues we had...

**Pull request merged**